### PR TITLE
Credential Registry testcases, help page testcases,    https://mosip.atlassian.net/browse/MOSIP-30054

### PR DESCRIPTION
**Pull request by anup-nehe**
https://mosip.atlassian.net/browse/MOSIP-30282
https://mosip.atlassian.net/browse/MOSIP-30054
https://mosip.atlassian.net/browse/MOSIP-30230
added new test cases:
-Credential Registry testcases.
-valid context in the help page.
-accessing help page in offline.
-scrolling the help page.
-deleteDownloadedVcInOfflineMode.

### CODE REVIEW

**Review by lsivanand — APPROVED** [review]

File: injitest/src/main/java/io/mosip/test/mob/inji/api/BaseTestCase.java [hunk 1]
```diff
@@ -63,6 +65,10 @@ public static void intiateUINGenration() {
 			perpetualVid = AdminTestUtil.generateVID(uin, "perpetual");
 			onetimeuseVid = AdminTestUtil.generateVID(uin, "onetimeuse");
 			temporaryVid= AdminTestUtil.generateVID(uin, "temporary");
+			
+			mockSMTPListener = new MockSMTPListener();
+			mockSMTPListener.run();
+
```
Comment (kamalsinghthoughtworks) [inline injitest/src/main/java/io/mosip/test/mob/inji/api/BaseTestCase.java:71]: extra spaces

File: injitest/src/main/java/io/mosip/test/mob/inji/pages/AboutInjiPage.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    private WebElement aboutInjiHeader;
+    
+    @AndroidFindBy(uiAutomator = "new UiSelector().textContains(\"Copy\")")
+    private WebElement Copy;
+    
+    @AndroidFindBy(uiAutomator = "new UiSelector().textContains(\"Copied\")")
+    private WebElement Copied;
+    
+    @AndroidFindBy(xpath = "//*[@resource-id=\"iconIcon\"]")
+    private WebElement Back;
```
Comment (kamalsinghthoughtworks) [inline injitest/src/main/java/io/mosip/test/mob/inji/pages/AboutInjiPage.java:18]: use uiAutomator

File: injitest/src/main/java/io/mosip/test/mob/inji/pages/AboutInjiPage.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+    public boolean isaboutInjiHeaderDisplayed() {
+        return this.isElementDisplayed(aboutInjiHeader, "ABOUT INJI");
+    }
+    
+    public boolean isAppidIsCopied() {
+        return this.isElementDisplayed(Copied, "Copied");
+    }
+    
+    public boolean isAppidIsCopy() {
```
Comment (kamalsinghthoughtworks) [inline injitest/src/main/java/io/mosip/test/mob/inji/pages/AboutInjiPage.java:32]: isAppidIsCopy() >> isCopyTextDisplayed()

File: injitest/src/main/java/io/mosip/test/mob/inji/pages/CredentialRegistryPage.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
+package io.mosip.test.mob.inji.pages;
+
+import io.appium.java_client.AppiumDriver;
+import io.appium.java_client.pagefactory.AndroidFindBy;
+import org.openqa.selenium.WebElement;
+
+public class CredentialRegistryPage extends BasePage {
+
+	@AndroidFindBy(xpath = "//android.widget.TextView[@text=\"Edit Credential Registry\"]")
```
Comment (kamalsinghthoughtworks) [inline injitest/src/main/java/io/mosip/test/mob/inji/pages/CredentialRegistryPage.java:9]: use uiAutomator

File: injitest/src/main/java/io/mosip/test/mob/inji/pages/CredentialRegistryPage.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    public WebElement CredentialRegistryTextBoxHeader;
+	
+	@AndroidFindBy(xpath = "(//android.widget.EditText[@resource-id=\"RNE__Input__text-input\"])[1]")
```
Comment (kamalsinghthoughtworks) [inline injitest/src/main/java/io/mosip/test/mob/inji/pages/CredentialRegistryPage.java:12]: use uiAutomator

File: injitest/src/main/java/io/mosip/test/mob/inji/pages/AboutInjiPage.java [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+        return this.isElementDisplayed(Copy, "Copy");
+    }
+
+    public AboutInjiPage clickOnCopy(){
+        clickOnElement(Copy);
+        return new AboutInjiPage(driver);
```
Comment (kamalsinghthoughtworks) [inline injitest/src/main/java/io/mosip/test/mob/inji/pages/AboutInjiPage.java:38]: return this;

File: injitest/src/main/java/io/mosip/test/mob/inji/pages/AboutInjiPage.java [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+    }
+    
+    public AboutInjiPage clickOnBack(){
+        clickOnElement(Copy);
+        return new AboutInjiPage(driver);
```
Comment (kamalsinghthoughtworks) [inline injitest/src/main/java/io/mosip/test/mob/inji/pages/AboutInjiPage.java:43]: return this;

File: injitest/src/main/java/io/mosip/test/mob/inji/pages/CredentialRegistryPage.java [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+    public CredentialRegistryPage(AppiumDriver driver) {
+        super(driver);
+    }
+    
+    public boolean isCredentialRegistryTextBoxHeaderDisplayed() {
+        return this.isElementDisplayed(CredentialRegistryTextBoxHeader, "Credential Registry Text heder is visible");
+    }
+    
+    public CredentialRegistryPage setEnterIdTextBox(String env) {
+    	CredentialRegistryTextBox.clear();
```
Comment (kamalsinghthoughtworks) [inline injitest/src/main/java/io/mosip/test/mob/inji/pages/CredentialRegistryPage.java:33]: create a method in BasePage class >> clearTextBoxAndSendKeys(String text)

File: injitest/src/main/java/io/mosip/test/mob/inji/pages/DetailedVcViewPage.java [hunk 9]
```diff
@@ -43,8 +43,18 @@ public class DetailedVcViewPage extends BasePage{
     @AndroidFindBy(accessibility = "profileAuthenticated")
     private WebElement profileAuthenticated;
 
-    @AndroidFindBy(accessibility = "close")
```
Comment (kamalsinghthoughtworks) [inline injitest/src/main/java/io/mosip/test/mob/inji/pages/DetailedVcViewPage.java:46]: move out to correct page class

File: injitest/src/main/java/io/mosip/test/mob/inji/pages/DetailedVcViewPage.java [hunk 10]
```diff
@@ -43,8 +43,18 @@ public class DetailedVcViewPage extends BasePage{
...
+//    @AndroidFindBy(accessibility = "close")
+//    private WebElement crossIcon;
+//    
+    @AndroidFindBy(xpath = "//*[@resource-id=\"iconIcon\"]")
+    @iOSXCUITFindBy(iOSClassChain = "**/XCUIElementTypeOther[`label == \"\uE5CD\"`][2]")
     private WebElement crossIcon;
+    
+    @AndroidFindBy(xpath = "//android.widget.ScrollView/android.view.ViewGroup/android.view.ViewGroup/android.view.ViewGroup/android.view.ViewGroup[1]/android.view.ViewGroup[1]/android.view.ViewGroup[1]/android.view.ViewGroup[1]/android.view.ViewGroup[1]/android.view.ViewGroup/android.view.ViewGroup")
+    private WebElement detailedVcViewPageQr;
```
Comment (kamalsinghthoughtworks) [inline injitest/src/main/java/io/mosip/test/mob/inji/pages/DetailedVcViewPage.java:54]: need to add test id in priority

File: injitest/src/main/java/io/mosip/test/mob/inji/pages/HelpPage.java [hunk 11]
```diff
@@ -4,8 +4,11 @@
 import io.appium.java_client.pagefactory.AndroidFindBy;
 import io.appium.java_client.pagefactory.iOSXCUITFindBy;
 
+import org.openqa.selenium.JavascriptExecutor;
```
Comment (kamalsinghthoughtworks) [inline injitest/src/main/java/io/mosip/test/mob/inji/pages/HelpPage.java:7]: remove unused imports

File: injitest/src/main/java/io/mosip/test/mob/inji/pages/MoreOptionsPage.java [hunk 12]
```diff
@@ -28,6 +28,9 @@ public class MoreOptionsPage extends BasePage {
     
     @AndroidFindBy(xpath = "//*[@resource-id=\"iconIcon\"]")
     private WebElement CloseButton;
+    
+    @AndroidFindBy(xpath = "//*[contains(@text,'Something is wrong. Please try again later!')]")
+    public WebElement SomthingIsWrongPopup;
```
Comment (kamalsinghthoughtworks) [inline injitest/src/main/java/io/mosip/test/mob/inji/pages/MoreOptionsPage.java:33]: use uiAutomator

File: injitest/src/main/java/io/mosip/test/mob/inji/pages/OtpVerificationPage.java [hunk 13]
```diff
@@ -45,6 +48,10 @@ public boolean invalidOtpMessageDisplayed() {
     public boolean somethingWetWrongInVcActivationDisplayed() {
         return this.isElementDisplayed(invalidOtpMessageInVcActivation, "Something is wrong. Please try again later!");
     }
+    
+    public boolean VidNotAvailableDisplayed() {
```
Comment (kamalsinghthoughtworks) [inline injitest/src/main/java/io/mosip/test/mob/inji/pages/OtpVerificationPage.java:52]: method camel casing

File: injitest/src/main/java/io/mosip/test/mob/inji/pages/ReceiveCardPage.java [hunk 14]
```diff
@@ -0,0 +1,50 @@
...
+import io.appium.java_client.pagefactory.AndroidFindBy;
+import io.appium.java_client.pagefactory.iOSXCUITFindBy;
+import org.openqa.selenium.WebElement;
+
+public class ReceiveCardPage extends BasePage {
+	 @AndroidFindBy(uiAutomator = "new UiSelector().textContains(\"Allow\")")
+	  private WebElement allowButton;
+	 
+	 @AndroidFindBy(uiAutomator = "new UiSelector().textContains(\"Display this QR code to request resident Card\")")
+	  private WebElement ReceiveCardHeder;
```
Comment (kamalsinghthoughtworks) [inline injitest/src/main/java/io/mosip/test/mob/inji/pages/ReceiveCardPage.java:14]: camel case

File: injitest/src/main/java/io/mosip/test/mob/inji/pages/ReceiveCardPage.java [hunk 15]
```diff
@@ -0,0 +1,50 @@
...
+	 
+	 @AndroidFindBy(uiAutomator = "new UiSelector().textContains(\"Ipakita ang QR code na ito para humiling ng resident card\")")
+	  private WebElement ReceiveCardHederInFilipinoLanguage;
```
Comment (kamalsinghthoughtworks) [inline injitest/src/main/java/io/mosip/test/mob/inji/pages/ReceiveCardPage.java:17]: camel case

File: injitest/src/main/java/io/mosip/test/mob/inji/pages/ReceiveCardPage.java [hunk 16]
```diff
@@ -0,0 +1,50 @@
...
+	 
+	 @AndroidFindBy(xpath = "//android.view.ViewGroup/descendant::android.view.ViewGroup[last()]")
+	  private WebElement QRCode;
```
Comment (kamalsinghthoughtworks) [inline injitest/src/main/java/io/mosip/test/mob/inji/pages/ReceiveCardPage.java:20]: camel case

File: injitest/src/main/java/io/mosip/test/mob/inji/pages/ReceiveCardPage.java [hunk 17]
```diff
@@ -0,0 +1,50 @@
...
+	 @AndroidFindBy(uiAutomator = "new UiSelector().textContains(\"Waiting for connection...\")")
+	  private WebElement watitingForConnection;
+
+    public ReceiveCardPage(AppiumDriver driver) {
+        super(driver);
+    }
+    
+    public ReceiveCardPage ClickOnAllowButton() {
+    	 this.clickOnElement(allowButton);
+         return new ReceiveCardPage(driver);
```
Comment (kamalsinghthoughtworks) [inline injitest/src/main/java/io/mosip/test/mob/inji/pages/ReceiveCardPage.java:31]: return this;

File: injitest/src/main/java/io/mosip/test/mob/inji/pages/ReceiveCardPage.java [hunk 18]
```diff
@@ -0,0 +1,50 @@
...
+    
+    public boolean isReceiveCardHederDisplayed() {
+        return this.isElementDisplayed(ReceiveCardHeder, "Display this QR code to request resident Card");
+    }
+    
+    public boolean isReceiveCardHederInFilipinoLanguageDisplayed() {
+        return this.isElementDisplayed(ReceiveCardHederInFilipinoLanguage, "Ipakita ang QR code na ito para humiling ng resident card");
+    }
+    
+    public boolean isWaitingForconnectionDisplayed() {
```
Comment (kamalsinghthoughtworks) [inline injitest/src/main/java/io/mosip/test/mob/inji/pages/ReceiveCardPage.java:42]: Camel Case

File: injitest/src/main/java/io/mosip/test/mob/inji/pages/SettingsPage.java [hunk 19]
```diff
@@ -26,7 +26,28 @@ public class SettingsPage extends BasePage {
     @AndroidFindBy(xpath = "//*[contains(@text,'Wika')]")
     @iOSXCUITFindBy(iOSClassChain = "**/XCUIElementTypeStaticText[`label == \"Wika\"`]")
     private WebElement wikaButton;
-
+    
+    @AndroidFindBy(xpath = "//*[contains(@text,'Inji Tour Guide')]")
+    public WebElement InjiTourGuide;
```
Comment (kamalsinghthoughtworks) [inline injitest/src/main/java/io/mosip/test/mob/inji/pages/SettingsPage.java:31]: camel case

File: injitest/src/main/java/io/mosip/test/mob/inji/pages/SettingsPage.java [hunk 20]
```diff
@@ -26,7 +26,28 @@ public class SettingsPage extends BasePage {
...
+    
+    @AndroidFindBy(xpath = "//*[contains(@text,'Credential Registry')]")
+    public WebElement ClickOnCredentialRegistry;
```
Comment (kamalsinghthoughtworks) [inline injitest/src/main/java/io/mosip/test/mob/inji/pages/SettingsPage.java:34]: camel case

File: injitest/src/main/java/io/mosip/test/mob/inji/pages/SettingsPage.java [hunk 21]
```diff
@@ -26,7 +26,28 @@ public class SettingsPage extends BasePage {
...
+    
+    @AndroidFindBy(xpath = "//*[contains(@text,'Receive Card')]")
+    public WebElement ClickOnReceiveCard;
```
Comment (kamalsinghthoughtworks) [inline injitest/src/main/java/io/mosip/test/mob/inji/pages/SettingsPage.java:37]: camel case

File: injitest/src/main/java/io/mosip/test/mob/inji/pages/SettingsPage.java [hunk 22]
```diff
@@ -26,7 +26,28 @@ public class SettingsPage extends BasePage {
...
+    
+    @AndroidFindBy(xpath = "//*[contains(@text,'Tumanggap ng Card')]")
+    public WebElement ClickOnReceiveCardInfilipinoLanguage;
```
Comment (kamalsinghthoughtworks) [inline injitest/src/main/java/io/mosip/test/mob/inji/pages/SettingsPage.java:40]: camel case

File: injitest/src/main/java/io/mosip/test/mob/inji/pages/SettingsPage.java [hunk 23]
```diff
@@ -26,7 +26,28 @@ public class SettingsPage extends BasePage {
...
+    
+    @AndroidFindBy(uiAutomator = "new UiSelector().textContains(\"About Inji\")")
+    private WebElement AboutInji;
```
Comment (kamalsinghthoughtworks) [inline injitest/src/main/java/io/mosip/test/mob/inji/pages/SettingsPage.java:43]: camel case

File: injitest/src/main/java/io/mosip/test/mob/inji/pages/WelcomePage.java [hunk 24]
```diff
@@ -22,7 +22,9 @@ public class WelcomePage extends BasePage {
     @AndroidFindBy(accessibility = "next")
     @iOSXCUITFindBy(iOSClassChain = "**/XCUIElementTypeOther[`label == \"Next\"`][4]")
     private WebElement nextButton;
-
+    
+    @AndroidFindBy(xpath = "//*[contains(@text,'Back')]")
+    public WebElement ClickBack;
```
Comment (kamalsinghthoughtworks) [inline injitest/src/main/java/io/mosip/test/mob/inji/pages/WelcomePage.java:27]: backButton

File: injitest/src/main/java/io/mosip/test/mob/inji/testcases/AboutInjiTest.java [hunk 25]
```diff
@@ -0,0 +1,46 @@
...
+        assertTrue(settingsPage.isSettingPageLoaded(), "Verify if setting page is displayed");
+        AboutInjiPage aboutInjiPage = settingsPage.clickOnAbouInji();
+        
+        assertTrue(aboutInjiPage.isaboutInjiHeaderDisplayed(),"Verify id about inji page displayed");
+        
+        aboutInjiPage.clickOnCopy();
+        assertTrue(aboutInjiPage.isAppidIsCopied(),"verify if app id is copied");
+        
+        aboutInjiPage.clickOnBack();
+        assertTrue(aboutInjiPage.isAppidIsCopy(),"verify if app id is copy");
```
Comment (kamalsinghthoughtworks) [inline injitest/src/main/java/io/mosip/test/mob/inji/testcases/AboutInjiTest.java:42]: message need to update

**Pull request closed without merging**